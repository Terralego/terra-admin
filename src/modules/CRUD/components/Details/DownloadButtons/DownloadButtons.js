import React from 'react';
import PropTypes from 'prop-types';
import { AnchorButton, ButtonGroup } from '@blueprintjs/core';

const DownloadButtons = ({ documents, ...rest }) => documents.length > 0 && (
  <ButtonGroup {...rest}>
    {documents.map(({ download_url: url, template_file: file, template_name: name }) => (
      <AnchorButton key={file} icon="download" href={url}>{name}</AnchorButton>
    ))}
  </ButtonGroup>
);

DownloadButtons.propTypes = {
  documents: PropTypes.arrayOf(
    PropTypes.shape({
      template_name: PropTypes.string,
      download_url: PropTypes.string,
      template_file: PropTypes.string,
    }),
  ),
};

DownloadButtons.defaultProps = {
  documents: [],
};

export default DownloadButtons;
