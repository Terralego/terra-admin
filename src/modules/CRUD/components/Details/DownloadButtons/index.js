import { withNamespaces } from 'react-i18next';

import compose from '../../../../../utils/compose';
import DownloadButtons from './DownloadButtons';

export default compose(
  withNamespaces(),
)(DownloadButtons);
