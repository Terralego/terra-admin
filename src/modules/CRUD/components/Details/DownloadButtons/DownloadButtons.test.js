import React from 'react';
import renderer from 'react-test-renderer';

import DownloadButtons from './DownloadButtons';

jest.mock('@blueprintjs/core', () => ({
  ButtonGroup: ({ children }) => <ul>{children}</ul>,
  AnchorButton: ({ children }) => <li>{children}</li>,
}));


const props = {
  files: [{
    name: 'FileName',
    url: 'path/to/file/{id}',
  }, {
    name: 'FileName2',
    url: 'path/to/file2/{id}',
  }],
  id: 3,
};


it('should render correctly', () => {
  const tree = renderer.create((
    <DownloadButtons
      {...props}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
