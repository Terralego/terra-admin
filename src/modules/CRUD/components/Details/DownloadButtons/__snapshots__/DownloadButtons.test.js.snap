// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<ul>
  <li>
    FileName
  </li>
  <li>
    FileName2
  </li>
</ul>
`;
